src/VisualStudio/Xaml/Impl/XamlStaticTypeDefinitions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Xaml
+{
+    public static class XamlStaticTypeDefinitions
+    {
+        [Export]
+        [Name(ContentTypeNames.XamlContentType)]
+        [BaseDefinition(ContentTypeNames.RoslynContentType)]
+        public static readonly ContentTypeDefinition XamlContentTypeDefinition;
+
+        // Associate .xaml as the Xaml content type.
```

Comment:
We certainly don't have the association of XamlContentType => RoslynContentType anywhere else. As for the extension is concerned, I'd rather be explicit than rely on a more obscure mechanism.